src/energyplus/ForwardTranslator/ForwardTranslatePlantLoop.cpp

```diff
@@ -334,21 +339,45 @@ namespace energyplus {
             auto sourceLoop = central_hp->sourcePlantLoop();
 
             // supply = cooling Loop
-            if (loop.handle() == coolingLoop->handle()) {
+            if (coolingLoop && loop.handle() == coolingLoop->handle()) {
               inletNode = waterToWaterComponent->supplyInletModelObject()->optionalCast<Node>();
               outletNode = waterToWaterComponent->supplyOutletModelObject()->optionalCast<Node>();
 
               // tertiary = heating loop
-            } else if (loop.handle() == heatingLoop->handle()) {
+            } else if (heatingLoop && loop.handle() == heatingLoop->handle()) {
               inletNode = waterToWaterComponent->tertiaryInletModelObject()->optionalCast<Node>();
               outletNode = waterToWaterComponent->tertiaryOutletModelObject()->optionalCast<Node>();
 
               // demand = source loop
-            } else if (loop.handle() == sourceLoop->handle()) {
+            } else if (sourceLoop && loop.handle() == sourceLoop->handle()) {
               inletNode = waterToWaterComponent->demandInletModelObject()->optionalCast<Node>();
               outletNode = waterToWaterComponent->demandOutletModelObject()->optionalCast<Node>();
             }
 
+          } else if (auto ch = modelObject.optionalCast<ChillerElectricASHRAE205>()) {
+            // This one has **FIVE** loops and can be potentially several time on the same loop (demand side, eg: Condenser + Oil Cooler + Auxiliary)
+
+            // Doing this: `if (ch->chilledWaterLoop() && loop.handle() == ch->chilledWaterLoop()->handle())` isn't enough because we could be on the
+            // same loop, so we rely on prevNode_ instead
+            OS_ASSERT(prevNode_);
+
+            if (ch->chilledWaterInletNode() && ch->chilledWaterInletNode()->handle() == prevNode_->handle()) {
+              inletNode = ch->chilledWaterInletNode();
+              outletNode = ch->chilledWaterOutletNode();
+            } else if (ch->condenserInletNode() && ch->condenserInletNode()->handle() == prevNode_->handle()) {
+              inletNode = ch->condenserInletNode();
+              outletNode = ch->condenserOutletNode();
+            } else if (ch->heatRecoveryInletNode() && ch->heatRecoveryInletNode()->handle() == prevNode_->handle()) {
+              inletNode = ch->heatRecoveryInletNode();
+              outletNode = ch->heatRecoveryOutletNode();
+            } else if (ch->oilCoolerInletNode() && ch->oilCoolerInletNode()->handle() == prevNode_->handle()) {
+              inletNode = ch->oilCoolerInletNode();
+              outletNode = ch->oilCoolerOutletNode();
+            } else if (ch->auxiliaryInletNode() && ch->auxiliaryInletNode()->handle() == prevNode_->handle()) {
+              inletNode = ch->auxiliaryInletNode();
+              outletNode = ch->auxiliaryOutletNode();
+            }
```

Comment:
PopulateBranch: check the prev node to identify the proper inlet/outlet nodes we're dealing with. Technically I could have this component on **FOUR** separate demand branches on the **SAME** loop.